libs/hdf-converters/src/asff-mapper/case-hdf2asff.ts

```diff
@@ -154,6 +154,40 @@ function getCodeForProfileLayer(
   }
 }
 
+function getPassthrough(execTypes: Record<string, unknown>) {
+  //Default behavior for old instances of HDF
+  if (_.has(execTypes, 'Execution.passthrough')) {
+    return _.get(execTypes, 'Execution.passthrough');
+  }
+  //If only one field exists, then is already an object
+  if (_.has(execTypes, 'Execution.passthrough1of1')) {
+    return _.get(execTypes, 'Execution.passthrough1of1');
+  }
+  //Else reconstitute passthrough from detected strings
+  const keysArr = [];
+  const strArr = [];
+  for (const key of _.keys(execTypes.Execution)) {
+    if (/passthrough.*of/.test(key)) {
+      keysArr.push(_.pick(execTypes.Execution, key));
+    }
+  }
+  const cntMax = keysArr.length;
+  let cntMin = 1;
+  //No valid passthrough
+  if (cntMax <= 1) {
+    return {};
+  }
+  while (cntMin <= cntMax) {
+    for (const obj of keysArr) {
+      if ((_.keys(obj)[0].match(/\d+/) || '')[0] === String(cntMin)) {
+        strArr.push(_.values(obj)[0]);
+      }
+      cntMin++;
+    }
+  }
+  return JSON.parse(strArr.join(''));
```

Comment:
this will throw an error if the json is not parsable.  we need some way to handle that here.  we should either not return anything at all for passthrough or only provide an incomplete string.